tests/TestRunner.NUnit/NUnitTestRunner.cs

```diff
@@ -0,0 +1,202 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+using Android.Content;
+using Android.OS;
+using Android.Util;
+
+using NUnitLite.Runner;
+using NUnit.Framework.Api;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.WorkItems;
+using NUnit.Framework.Internal.Filters;
+
+using NUnitTest = NUnit.Framework.Internal.Test;
+
+namespace Xamarin.Android.UnitTests.NUnit
+{
+	public class NUnitTestRunner : TestRunner, ITestListener
+	{
+		Dictionary<string, object> builderSettings;
+		TestSuiteResult results;
+
+		public ITestFilter Filter { get; set; }
+		public bool GCAfterEachFixture { get; set; }
+
+		protected override string ResultsFileName { get; set; } = "TestResults.NUnit.xml";
+
+		public NUnitTestRunner (Context context, LogWriter logger, Bundle bundle) : base (context, logger, bundle)
+		{
+			builderSettings = new Dictionary<string, object> (StringComparer.OrdinalIgnoreCase);
+		}
+
+		public override void Run (IList<TestAssemblyInfo> testAssemblies)
+		{
+			if (testAssemblies == null)
+				throw new ArgumentNullException (nameof (testAssemblies));
+			
+			var builder = new NUnitLiteTestAssemblyBuilder ();
+			var runner = new NUnitLiteTestAssemblyRunner (builder);
+			var testSuite = new TestSuite (global::Android.App.Application.Context.PackageName);
+			results = new TestSuiteResult (testSuite);
+
+			foreach (TestAssemblyInfo assemblyInfo in testAssemblies) {
+				if (assemblyInfo == null || assemblyInfo.Assembly == null)
+					continue;
+				
+				if (!runner.Load (assemblyInfo.Assembly, builderSettings)) {
+					OnWarning ($"Failed to load tests from assembly '{assemblyInfo.Assembly}");
+					continue;
+				}
+				if (runner.LoadedTest is NUnitTest tests)
+					testSuite.Add (tests);
+				
+				// Messy API. .Run returns ITestResult which is, in reality, an instance of TestResult since that's
+				// what WorkItem returns and we need an instance of TestResult to add it to TestSuiteResult. So, cast
+				// the return to TestResult and hope for the best.
+				ITestResult result = null;
+				try {
+					OnAssemblyStart (assemblyInfo.Assembly);
+					result = runner.Run (this, Filter);
+				} finally {
+					OnAssemblyFinish (assemblyInfo.Assembly);
+				}
+
+				if (result == null)
+					continue;
+
+				var testResult = result as TestResult;
+				if (testResult == null)
+					throw new InvalidOperationException ($"Unexpected test result type '{result.GetType ()}'");
+				results.AddResult (testResult);
+			}
+
+			LogFailureSummary ();
+		}
+
+		public bool Pass (ITest test)
+		{
+			return true;
+		}
+
+		public void TestFinished (ITestResult result)
+		{
+			if (result.Test is TestSuite) {
+				//if (!result.IsError && !result.IsFailure && !result.IsSuccess && !result.Executed)
+				//Writer.WriteLine ("\t[INFO] {0}", result.Message);
+				if (result.ResultState.Status != TestStatus.Failed &&
+				    result.ResultState.Status != TestStatus.Skipped &&
```

Comment:
The indentation here is wonky/inconsistent.

Expressions within `if`/`while`/etc. should be indented one tab stop *further* than the *contents* of the `if`/`while`/etc.

Indentation should be consistent on every line.

Thus:

```csharp
if (result.ResultState.Status != TestStatus.Failed &&
        result.ResultState.Status != TestStatus.Skipped &&
        result.ResultState.Status != TestStatus.Passed &&
        result.ResultState.Status != TestStatus.Inconclusive) {
    Logger.OnInfo ("\t[INFO] {0}", result.Message);
}
```